### PR TITLE
Fix `detci` printing

### PR DESCRIPTION
**Pull request by philipmnel**
## Description
Fixes issues #2487 where the `print` keyword was not being properly set in part of `detci` module. S^2 value now prints out if `calc_s_squared` is turned on.

## Checklist
- [x] CI tests pass

## Status
- [x] Ready for review
- [x] Ready for merge

### CODE REVIEW

File: psi4/src/psi4/detci/params.cc [hunk 1]
```diff
@@ -74,7 +74,7 @@ void CIWavefunction::get_parameters(Options &options) {
     Parameters_->wfn = options.get_str("WFN");
 
     // CDS-TODO: Check these
-    print_ = 1;
+    Parameters_->print_ = 1;
```
Comment (JonathonMisiewicz) [inline psi4/src/psi4/detci/params.cc:77]: Why is this needed? I would _expect_ that `CIWavefunction` is a wavefunction subclass and thus inherits the wavefunction's `print_` field. It doesn't need additional "parameters".

**Comment by JonathonMisiewicz** [comment]
Thanks for the PR! I want to make sure we're fixing #2487 the right way before merging this in.

File: psi4/src/psi4/detci/params.cc (hunk 1 above)
Comment (philipmnel) [reply psi4/src/psi4/detci/params.cc:77]: I added `print_` to the parameters struct so that it can be passed to `CIVect`, where the un-set `print_lvl` used to live. `CIVect` gets passed the parameters struct, but not the wavefunction.

**Review by JonathonMisiewicz — APPROVED** [review]

**Review by loriab — APPROVED** [review]
thanks for the fix!
